connectors/mongo/conn.go

```diff
@@ -376,30 +401,49 @@ func (c *conn) ListData(ctx context.Context, r *connect.Request[adiomv1.ListData
 		}
 	}
 	c.buffersMutex.RLock()
-	ch, ok := c.buffers[cursorID]
+	buffer, ok := c.buffers[cursorID]
 	c.buffersMutex.RUnlock()
 	if !ok {
 		return nil, connect.NewError(connect.CodeInvalidArgument, fmt.Errorf("invalid cursor"))
 	}
+	if ctr+1 == buffer.ctr && buffer.last != nil { // Handle repeated requests for same page
+		buffer.lastUsed = time.Now()
```

Comment:
so even though we update this, we never reset the timer so we might end up in a case where we never cleanup